tic-tac-toe.js

```diff
@@ -1,188 +1,172 @@
-let players = ["player1", "player2"];
-let states = ["open", "player1won", "player2won", "draw"];
+const Marks = {
+  P1: "X",
+  P2: "O",
+  Empty: ".",
+};
+
+const players = {
+  P1: "player1",
+  P2: "player2",
+};
+
+const states = {
+  open: "open",
+  P1: "player1won",
+  P2: "player2won",
+  dran: "draw",
+};
+
+/*const players = ["player1", "player2"];
+const states = ["open", "player1won", "player2won", "draw"];
+const squareStates = ["not set", "player 1", "player 2"];*/
 
-class TicTacToe {
-    constructor(){
+export class TicTacToe {
+  // Saninn's comment: Usually the functions should be defined AFTER the constructor... but ok...
+  resetGame() {
     this.field = [
-        [".", ".", "."],
-        [".", ".", "."],
-        [".", ".", "."]
-        ];
-    this.squareState = ["not set", "player 1", "player 2"];
-    /* I HAVEN'T USED THIS*/
-    this.currentPlayer = players[0];
-    this.currentState = states[0];
-    this.movesArray = [];
-    };
-    
-    resetGame() {
-        this.field = [
-            [".", ".", "."],
-            [".", ".", "."],
-            [".", ".", "."]
-            ];
-        this.currentPlayer = players[0];
-        this.currentState = states[0];
+      [Marks.Empty, Marks.Empty, Marks.Empty],
+      [Marks.Empty, Marks.Empty, Marks.Empty],
+      [Marks.Empty, Marks.Empty, Marks.Empty],
+    ];
+    this.currentPlayer = players.P1;
+    this.currentState = states.open;
+  }
+
+  constructor() {
+    this.resetGame();
+  }
+
+  _playerHasWon(userMark) {
```

Comment:
I have the feeling we choose a non-so-good name with `_playerHasWon` because the parameter it takes is not a `player` (from the `players` enum) but it takes a `marker` as param.

It would be cool if we can somehow unify `player` and `marker`.

Reply:
Uhmmmmmm don't really see the problem.